examples/wasip3-competing-outbound-http-calls/src/lib.rs

```diff
@@ -0,0 +1,121 @@
+use std::pin::pin;
+
+use futures::SinkExt;
+use http::Request;
+use serde::{Deserialize, Serialize};
+use spin_sdk::http_wasip3::http_service;
+use spin_sdk::http_wasip3::{send, EmptyBody, IntoResponse};
+
+use bytes::Bytes;
+use futures::{
+    channel::mpsc::{channel, Sender},
+    StreamExt,
+};
+
+#[http_service]
+async fn handle_competing_outbound_http_calls(
+    _req: spin_sdk::http_wasip3::Request,
+) -> impl IntoResponse {
+    println!("Handling reuqest");
+
+    // A lot of code taken from: https://github.com/spinframework/spin-rust-sdk/blob/main/examples/wasip3-streaming/src/lib.rs
+
+    // Create a streaming Body implementation that backs onto a `mpsc`
+    // channel. The function returns the sender side of the channel; the
+    // receiver end becomes the body. So anything written to the sender
+    // side will be sent out over the HTTP response.
+    let (mut tx, body) = bytes_stream_body();
+
+    // Use wit_bindgen::spawn to allow the async block to keep running
+    // after the handler returns.
+    spin_sdk::http_wasip3::wasip3::wit_bindgen::spawn(async move {
+        // The two outbound calls
+        let spin = pin!(get_content_length("https://spinframework.dev"));
+        let book = pin!(get_content_length(
+            "https://component-model.bytecodealliance.org"
+        ));
+
+        // Getting the first response back
+        let first_result = futures::future::select(spin, book).await.factor_first();
+
+        // We need to keep the future around to also handle the second response,
+        // hence instantiating a new variable for the ContentLength struct
+        let first_response = first_result.0.unwrap();
+
+        // Let's print some stats to the server console
+        println!(
+            "HEAD request done: {:?}, took {:?} ms, content-length: {:?}",
+            first_response.url, first_response.ms_taken, first_response.content_length
+        );
+
+        // Serializing the struct as JSON represented as bytes
```

Comment:
```suggestion
        // Serialize the struct as JSON represented as bytes
```

I'm not sure what "represented as bytes" is trying to say here.  Everything is represented as bytes: is it alluding to the use of the `Bytes` struct.  In that case maybe "represented as a `Bytes`"?